File: matiasmasso/Mat.NET/DTO/DTOIrpf.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOIrpf
    {
        public DTOEmp Emp { get; set; }
        public DTOYearMonth YearMonth { get; set; }
        public List<Item> Items { get; set; }
        public List<DTOPgcSaldo> Saldos { get; set; }

        public DTOIrpf() : base()
        {
            Items = new List<Item>();
            Saldos = new List<DTOPgcSaldo>();
        }

        public static DTOIrpf Factory(DTOEmp oEmp, int year, int month)
        {
            DTOIrpf retval = new DTOIrpf();
            {
                var withBlock = retval;
                withBlock.Emp = oEmp;
                withBlock.YearMonth = new DTOYearMonth(year, (DTOYearMonth.Months)month);
            }
            return retval;
        }

        public DateTime Fch()
        {
            return YearMonth.LastFch();
        }

        public class Item
        {
            public DTOCcb Ccb { get; set; }
            public DTOAmt Base { get; set; }
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOIrpf.vb
Public Class DTOIrpf
    Property emp As DTOEmp
    Property yearMonth As DTOYearMonth
    Property items As List(Of Item)
    Property saldos As List(Of DTOPgcSaldo)

    Public Sub New()
        MyBase.New
        _Items = New List(Of Item)
        _Saldos = New List(Of DTOPgcSaldo)
    End Sub

    Shared Function Factory(oEmp As DTOEmp, year As Integer, month As Integer) As DTOIrpf
        Dim retval As New DTOIrpf
        With retval
            .emp = oEmp
            .yearMonth = New DTOYearMonth(year, month)
        End With
        Return retval
    End Function

    Public Function fch() As Date
        Return _yearMonth.LastFch
    End Function

    Public Class Item
        Property Ccb As DTOCcb
        Property Base As DTOAmt
    End Class


End Class
